crates/ra_mbe/src/syntax_bridge.rs

```diff
@@ -45,92 +45,65 @@ pub fn syntax_node_to_token_tree(node: &SyntaxNode) -> Option<(tt::Subtree, Toke
 //
 //
 
-/// Parses the token tree (result of macro expansion) to an expression
-pub fn token_tree_to_expr(tt: &tt::Subtree) -> Result<TreeArc<ast::Expr>, ExpandError> {
+fn token_tree_to_syntax_node<F>(tt: &tt::Subtree, f: F) -> Result<TreeArc<SyntaxNode>, ExpandError>
+where
+    F: Fn(&mut ra_parser::TokenSource, &mut ra_parser::TreeSink),
+{
     let buffer = tt::buffer::TokenBuffer::new(&[tt.clone().into()]);
     let mut token_source = SubtreeTokenSource::new(&buffer);
     let querier = token_source.querier();
     let mut tree_sink = TtTreeSink::new(querier.as_ref());
-    ra_parser::parse_expr(&mut token_source, &mut tree_sink);
+    f(&mut token_source, &mut tree_sink);
+    if tree_sink.roots.len() != 1 {
```

Comment:
The repetition of code. Below is such a code there

Reply:
Good catch, fixed!